xarray/core/variable.py

```diff
@@ -2590,12 +2590,14 @@ class IndexVariable(Variable):
     unless another name is given.
     """
 
-    __slots__ = ()
+    __slots__ = ("_name",)
```

Comment:
Can we remove the `name` property instead or is that much much harder?

Reply:
This would be better indeed but this is likely more work (not sure how much, though).

IndexVariable still needs a deeper refactor (#8124), or even be eventually dropped?

Reply:
After a quick check: internally only in a few places (conventions / multiindex check, dataarray creation, maybe groupby?) actually.

Externally I have no idea (IndexVariable is public API in theory).